sync/sync.go

```diff
@@ -289,52 +299,66 @@ Register:
 	}
 
 	logrus.Infof("PrepareRebuild %v", replicaAddress)
-	output, err := t.client.PrepareRebuild(rest.EncodeID(replicaAddress))
+	_, err = t.client.PrepareRebuild(rest.EncodeID(replicaAddress))
 	if err != nil {
 		return fmt.Errorf("failed to prepare rebuild, error: %s", err.Error())
 	}
 	inject.PanicAfterPrepareRebuild()
 
-	ok, err := t.isRevisionCountAndChainSame(fromClient, toClient)
+	ok, chain, err := t.isRevisionCountAndChainSame(fromClient, toClient)
 	if err != nil {
 		return err
 	}
 
 	if !ok {
 		logrus.Infof("syncFiles from:%v to:%v", fromClient, toClient)
-		if err = t.syncFiles(fromClient, toClient, output.Disks); err != nil {
+		if err = t.syncFiles(fromClient, toClient, chain); err != nil {
 			return err
 		}
 	}
-
 	logrus.Infof("reloadAndVerify %v", replicaAddress)
 	return t.reloadAndVerify(s, replicaAddress, toClient)
 }
 
-func (t *Task) isRevisionCountAndChainSame(fromClient, toClient *replicaClient.ReplicaClient) (bool, error) {
+func (t *Task) isRevisionCountAndChainSame(fromClient, toClient *replicaClient.ReplicaClient) (bool, []string, error) {
 	rwReplica, err := fromClient.GetReplica()
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
 
 	curReplica, err := toClient.GetReplica()
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
 	logrus.Infof("RevisionCount of RW replica (%v): %v, Current replica (%v): %v",
 		fromClient.GetAddress(), rwReplica.RevisionCounter, toClient.GetAddress(),
 		curReplica.RevisionCounter)
 	logrus.Infof("RW replica chain: %v, cur replica chain: %v", rwReplica.Chain, curReplica.Chain)
+
+	if curReplica.Checkpoint != "" {
```

Comment:
should we make sure checkpoints are also same?

Reply:
Here, while syncing it is not required and not necessary to be same. Suppose while creating checkpoint the replica crashed, both replicas will not have same checkpoint in that case.
What can be done over here is, data can be synced after the checkpoint of the RW Replica, since if RW Replica has a checkpoint that snapshot has to be present in WO replica also.
But for a replica from scratch, i.e, if a replica is not having checkpoint, whole data should be synced.